assignment7.txt

```diff
@@ -0,0 +1,19 @@
+Yael Kelmer. 
+
+1. ++*p increments the value of the datatype located at pointer p and then points at that location. *p++ points at the location in memory of p and then increments the value there.  
+
+2. The left to right order is guaranteed for operator precedence. 
+
+
+3. One advantage of pointers is that you can move along a string and point to different parts of it. 
+
+4.1 datatype: char array
+4.2 this is invalid
+4.3 datatype: int. value: 0
+4.4 datatype: int. value: 10
+4.5 datatype: int. value: 10
```

Comment:
Close! ```This actually returns the type int*. Recall that a[0] would be of type int and the ampersand just returns the memory address. So &a[0] would return the pointer to the address (since a pointer is basically an address, which is why all pointers are 8 bytes no matter what they're pointing to), which would be of type int*. ```